src/components/ShortenedLink.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { pallete } from "../misc/pallete";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faX } from "@fortawesome/free-solid-svg-icons";
+import Button from "./Button";
+import { motion } from "framer-motion";
+import PropTypes from "prop-types";
+const isLink = function (props, propName, componentName) {
+  const regex =
+    /[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/;
+
+  if (!regex.test(props[propName])) {
+    return new Error(
+      `Invalid prop ${propName} passed to ${componentName}. Expected a link.`
+    );
+  }
+};
+export default function ShortenedLink({ long, short, id, setLinkArr }) {
+  const [isCopied, setIsCopied] = useState(false);
+  const handleClick = () => {
+    navigator.clipboard.writeText(short);
```

Comment:
w teorii funkcja writeText zwraca Promisa, co w tym przypadku oznacza,ze ta operacja może się nie udać i wówczas ustawianie 'setIsCopied(true)' będzie błędne. W praktyce raczej są na to mniejsze szanse, ale warto się zabezpieczyć. Możesz skorzystać z funkcji 'then', aby wywować akcję, która ma mieć miejsce gdy Promise się wykona poprawnie (i właśnie w tej funkcji 'then()' ustawić setIsCopied(true)) i dodać też funkcję 'catch' która obsłuży przypadek, gdzie kopiowanie się nie powiedzie (Promise nie wykona się poprawnie)

Reply:
daj znać jak tam z Promisami stoisz, jak coś to wyjaśnie na spokojnie